DirectionsRoute#fromJson RuntimeException

Run into the following 💥 while testing

```
08-08 13:57:37.398 11915-11915/com.mapbox.services.android.navigation.testapp E/NativeMapView: Exception in MapView.OnMapChangedListener
    java.lang.RuntimeException: Failed to invoke public com.mapbox.geojson.Point() with no args
        at com.google.gson.internal.ConstructorConstructor$3.construct(ConstructorConstructor.java:113)
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:212)
        at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.read(TypeAdapterRuntimeTypeWrapper.java:41)
        at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(CollectionTypeAdapterFactory.java:82)
        at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(CollectionTypeAdapterFactory.java:61)
        at com.mapbox.api.directions.v5.models.AutoValue_RouteOptions$GsonTypeAdapter.read(AutoValue_RouteOptions.java:339)
        at com.mapbox.api.directions.v5.models.AutoValue_RouteOptions$GsonTypeAdapter.read(AutoValue_RouteOptions.java:35)
        at com.mapbox.api.directions.v5.models.AutoValue_DirectionsRoute$GsonTypeAdapter.read(AutoValue_DirectionsRoute.java:205)
        at com.mapbox.api.directions.v5.models.AutoValue_DirectionsRoute$GsonTypeAdapter.read(AutoValue_DirectionsRoute.java:30)
        at com.google.gson.Gson.fromJson(Gson.java:927)
        at com.google.gson.Gson.fromJson(Gson.java:892)
        at com.google.gson.Gson.fromJson(Gson.java:841)
        at com.google.gson.Gson.fromJson(Gson.java:813)
        at com.mapbox.api.directions.v5.models.DirectionsRoute.fromJson(DirectionsRoute.java:147)
        at com.mapbox.services.android.navigation.ui.v5.NavigationLauncher.extractRoute(NavigationLauncher.java:62)
        at com.mapbox.services.android.navigation.ui.v5.MapboxNavigationActivity.extractRoute(MapboxNavigationActivity.java:122)
        at com.mapbox.services.android.navigation.ui.v5.MapboxNavigationActivity.onNavigationReady(MapboxNavigationActivity.java:98)
        at com.mapbox.services.android.navigation.ui.v5.NavigationView.onMapReady(NavigationView.java:203)
        at com.mapbox.mapboxsdk.maps.MapView$MapCallback.onMapReady(MapView.java:1157)
        at com.mapbox.mapboxsdk.maps.MapView$MapCallback.onMapChanged(MapView.java:1142)
        at com.mapbox.mapboxsdk.maps.NativeMapView.onMapChanged(NativeMapView.java:900)
        at android.os.MessageQueue.nativePollOnce(Native Method)
        at android.os.MessageQueue.next(MessageQueue.java:323)
        at android.os.Looper.loop(Looper.java:135)
        at android.app.ActivityThread.main(ActivityThread.java:5417)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
     Caused by: java.lang.InstantiationException: Can't instantiate abstract class com.mapbox.geojson.Point
        at java.lang.reflect.Constructor.newInstance(Native Method)
        at com.google.gson.internal.ConstructorConstructor$3.construct(ConstructorConstructor.java:110)
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:212) 
        at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.read(TypeAdapterRuntimeTypeWrapper.java:41) 
        at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(CollectionTypeAdapterFactory.java:82) 
        at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(CollectionTypeAdapterFactory.java:61) 
        at com.mapbox.api.directions.v5.models.AutoValue_RouteOptions$GsonTypeAdapter.read(AutoValue_RouteOptions.java:339) 
        at com.mapbox.api.directions.v5.models.AutoValue_RouteOptions$GsonTypeAdapter.read(AutoValue_RouteOptions.java:35) 
        at com.mapbox.api.directions.v5.models.AutoValue_DirectionsRoute$GsonTypeAdapter.read(AutoValue_DirectionsRoute.java:205) 
        at com.mapbox.api.directions.v5.models.AutoValue_DirectionsRoute$GsonTypeAdapter.read(AutoValue_DirectionsRoute.java:30) 
        at com.google.gson.Gson.fromJson(Gson.java:927) 
        at com.google.gson.Gson.fromJson(Gson.java:892) 
        at com.google.gson.Gson.fromJson(Gson.java:841) 
        at com.google.gson.Gson.fromJson(Gson.java:813) 
        at com.mapbox.api.directions.v5.models.DirectionsRoute.fromJson(DirectionsRoute.java:147) 
        at com.mapbox.services.android.navigation.ui.v5.NavigationLauncher.extractRoute(NavigationLauncher.java:62) 
        at com.mapbox.services.android.navigation.ui.v5.MapboxNavigationActivity.extractRoute(MapboxNavigationActivity.java:122) 
        at com.mapbox.services.android.navigation.ui.v5.MapboxNavigationActivity.onNavigationReady(MapboxNavigationActivity.java:98) 
        at com.mapbox.services.android.navigation.ui.v5.NavigationView.onMapReady(NavigationView.java:203) 
        at com.mapbox.mapboxsdk.maps.MapView$MapCallback.onMapReady(MapView.java:1157) 
        at com.mapbox.mapboxsdk.maps.MapView$MapCallback.onMapChanged(MapView.java:1142) 
        at com.mapbox.mapboxsdk.maps.NativeMapView.onMapChanged(NativeMapView.java:900) 
        at android.os.MessageQueue.nativePollOnce(Native Method) 
        at android.os.MessageQueue.next(MessageQueue.java:323) 
        at android.os.Looper.loop(Looper.java:135) 
        at android.app.ActivityThread.main(ActivityThread.java:5417) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616) 
```

Pretty easy to reproduce launching a navigation session using the `NavigationLauncher` 👉 [`NavigationLauncherActivity`](https://github.com/mapbox/mapbox-navigation-android/blob/master/app/src/main/java/com/mapbox/services/android/navigation/testapp/activity/navigationui/NavigationLauncherActivity.java)

I think introduced by the latest changes in MAS `3.4.0` https://github.com/mapbox/mapbox-navigation-android/pull/1196

--- a/gradle/dependencies.gradle
+++ b/gradle/dependencies.gradle
@@ -9,7 +9,7 @@ ext {
 
   version = [
       mapboxMapSdk       : '6.3.0',
-      mapboxSdkServices  : '3.4.0',
+      mapboxSdkServices  : '3.3.0',
       mapboxEvents       : '3.1.4',
       locationLayerPlugin: '0.7.1',
       autoValue          : '1.5.4',

